csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs

```diff
@@ -46,10 +46,9 @@ public static SessionOptions Default
             }
         }
 
-        private static SessionOptions MakeSessionOptionWithMklDnnProvider()
```

Comment:
Maybe we can make sessionoption register cpu provider on construction. And make SessionOptions only have 2 methods: RegisterCPUProvider and RegisterCudaProvider.

Reply:
the plan is to package the binary built without the --use_mkldnn option. Thus, providing the user option to register a provider that is not available seems problematic.

Reply:
the it makes sense to remove mkldnn.

Reply:
BTW, I will look into your refactoring suggestion as a separate work item.